src/world.rs

```diff
@@ -780,3 +794,66 @@ mod tests {
         assert!(world.read::<Pos>().get(e).is_some());
     }
 }
+
+#[cfg(feature="serialize")]
+/// Structure used to serialize a world using a component group.
+pub struct WorldSerializer<'a, G> {
+    world: &'a World,
+    phantom: PhantomData<G>,
+}
+
+#[cfg(feature="serialize")]
+impl<'a, G> WorldSerializer<'a, G> {
+    /// Creates a new world serializer out of a world.
+    pub fn new(world: &'a World) -> WorldSerializer<'a, G> {
+        WorldSerializer {
+            world: world,
+            phantom: PhantomData,
+        }
+    }
+}
+
+#[cfg(feature="serialize")]
+impl<'a, G> Serialize for WorldSerializer<'a, G>
+    where G: ComponentGroup + SerializeGroup
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer 
+    {
+        G::serialize_group(self.world, serializer)
+    }
+}
+
+#[cfg(feature="serialize")]
+/// Structure used for stateful deserialization into the world using a component group.
+pub struct WorldDeserializer<'a, G> {
+    world: &'a mut World,
+    entities: &'a [Entity],
+    phantom: PhantomData<G>,
+}
+
+#[cfg(feature="serialize")]
+impl<'a, G> WorldDeserializer<'a, G> {
+    /// Creates a new world deserializer out of a world and a list of entities.
+    ///
+    /// The list of entities will be used to merge into the component storages.
+    pub fn new(world: &'a mut World, entities: &'a [Entity]) -> WorldDeserializer<'a, G> {
```

Comment:
I wonder if it really makes sense to pass a slice of entities. Is there any case where you want to deserialize components and insert them for existing entities? I mean can you really ensure this works reliable? Which brings me to the question if it should be allowed for an entity to not have a component of the component group.